text/0028-ed25519-validation/0028-ed25519-validation.md

```diff
@@ -0,0 +1,79 @@
+
+Analogous to RFC 8032, the encoding of S _must_ represent an integer less than L.
+
+It is not possible for an external party to mutate R and still pass verification. The owner of the secret key, however, can create many different signatures for the same content: While Ed25519 defines a deterministic method of calculating the integer scalar r from the private key and the message, it is impossible to tell during signature verification if the point R = [r]B was created properly or any other scalar has been used.<br> As a result, there is a practically countless amount of different valid signatures corresponding to a certain message and public key.
+
+# Drawbacks
+
+- Allowing non-canonical encodings is a direct contradiction of RFC 8032 and rather unintuitive. Furthermore, it introduces alternative encodings for a handful of points on the curve. Even though such points will, for all practical purposes, never occur in honest signatures, it still theoretically introduces an external party malleability vector.
+- The cofactor validation is computationally slightly more expensive than the cofactor-less version since it requires a multiplication by 8.
+
+# Rationale and alternatives
+
+While the malleability of S poses serious issues and thus, must be prevented, the other two validation criteria, namely non-canonical encodings as well as the cofactor-less validation equation, could be relaxed without introducing attack vectors.
+
+Unfortunately, the Ed25519 `ref10` reference implementation as well as other implementations accept non-canonical points. As such, rejecting those inputs now would introduce a breaking change. While this might be acceptable for the IOTA protocol itself, since no Ed25519 signatures have been added to the ledger prior to this RFC, other consensus-critical applications require this backward compatibility with previously accepted signatures. Due to these considerations, the criterion was included in ZIP-215 to allow a seamless transition for existing consensus-critical contexts. This RFC aims to rather follow the existing ZIP-215 specification for compatibility and maintainability than to create a new standard.
+
+Using the cofactor-less validation poses a similar breaking change since signatures accepted by implementations using the cofactor validation would then be rejected. More importantly, however, in order to be able to use the much faster batch verification, the cofactor version is required. 
+
+# Unresolved questions
+
+- Are there issues resulting from the malleability introduced by non-canonical encodings of specially prepared public keys? 
+- Accepting non-canonical encodings of R allows 3rd party malleability of signatures, which is something that was previously explicitly prevented. Even though this should not effect honest signatures, is that really acceptable? 
```

Comment:
The probability of such R is extremely low. In case a 3rd party modifies y coordinate of point R the scalar k will change (assuming implementation uses byte representation of point R from signature bytes). So this doesn't really make signature malleable. Malicious signers can indeed include unreduced representation of R into signature.

Reply:
You are absolutely correct! I've just removed that unresolved questions :tada: when we make clear that the bytes from the signature is used (as stated above) everything is good